packages/terra-form/src/TextField.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Field from './Field';
+import Input from './Input';
+
+const propTypes = {
+  /**
+   * The defaultValue of the input element
+   */
+  defaultValue: PropTypes.string,
+  /**
+   * Error message for when the input is invalid
+   */
+  error: PropTypes.node,
+  /**
+   *  Help element to display with the input
+   */
+  help: PropTypes.node,
+  /**
+   * Custom attributes to apply to the input
+   */
+  inputAttrs: PropTypes.object,
+  /**
+   * Whether the field is inline
+   */
+  isInline: PropTypes.bool,
+  /**
+   * Label of the input
+   */
+  label: PropTypes.node,
+  /**
+   * Maximum number of characters user can input in this field
+   */
+  maxLength: PropTypes.number,
+  /**
+   * Minimum number of characters user must input in this field
+   */
+  minLength: PropTypes.number,
+  /**
+   * Name of the input attribute
+   */
+  name: PropTypes.string,
+  /**
+   * Function to trigger when user selects one of the radio buttons
+   */
+  onChange: PropTypes.func,
+  /**
+   * Whether the input is required
+   */
+  required: PropTypes.bool,
+  /**
+   * Type the input will be
+   */
+  type: PropTypes.string,
+  /**
+   * The value of the input element
+   */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  defaultValue: undefined,
+  error: null,
+  help: null,
+  inputAttrs: {},
+  isInline: false,
+  label: null,
+  maxLength: null,
+  minLength: null,
+  name: null,
+  onChange: undefined,
+  required: false,
+  type: 'text',
+  value: undefined,
+};
+
+const TextField = ({
+  defaultValue,
+  error,
+  help,
+  inputAttrs,
+  isInline,
+  label,
+  maxLength,
+  minLength,
+  name,
+  onChange,
+  required,
+  type,
+  value,
+  ...customProps
+}) => (
```

Comment:
I don't think we need to for these fields components. The idea was for all of them to extend off of the same field class, and all use the same styles. If we have a need to new styles, we can add them later, but if we add the ability for consumers to add their own class, that should suffice for our needs.